node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -618,6 +621,54 @@ struct State {
 	pending_candidates: HashMap<Hash, CollationEvent>,
 }
 
+/// Returns `true` if there's a leaf in our view close to the next group rotation
+/// and the validator will be assigned to core with a given para id.
```

Comment:
Is it only considered 'preconnect' if it's not for the current group?

We don't expect collators to be connected for the entire rotation (and we even kick them off if they're inactive too long)

Reply:
> Is it only considered 'preconnect' if it's not for the current group?

No, but we only perform this check if the para is not in our current assignment.